src/view/tables/I90Table.tsx

```diff
@@ -0,0 +1,230 @@
+import React from "react";
+import { CaseDetails } from "./CaseDetails";
+import { ChevronToggle } from "../util/ChevronToggle";
+import { connect } from "react-redux";
+import { Dispatch, AnyAction, bindActionCreators } from "redux";
+import { casesActionCreators, getCaseSummary } from "../../redux/modules/cases";
+import { RootState } from "../../redux/create";
+import { ELIS_CASE_BASE_URL, SNOOZE_OPTIONS } from "../../controller/config";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import ReactTooltip from "react-tooltip";
+import DateUtils from "../../utils/DateUtils";
+import NoteUtils from "../../utils/NoteUtils";
+import { SnoozeFormWrapper } from "../../controller/SnoozeFormWrapper";
+import { UpdateSnoozeFormWrapper } from "../../controller/UpdateSnoozeFormWrapper";
+import { appStatusActionCreators } from "../../redux/modules/appStatus";
+import "./ReceiptDisplayRow.scss";
+
+const mapStateToProps = (state: RootState) => ({
+  caselist: state.cases.caselist,
+  caseType: state.cases.type,
+  isLoading: state.cases.isLoading
+});
+
+const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
+  bindActionCreators(
+    {
+      toggleDetails: casesActionCreators.toggleDetails,
+      updateSummaryData: getCaseSummary,
+      setError: appStatusActionCreators.setDataLoadError,
+      setNotification: appStatusActionCreators.setNotification,
+      removeCase: casesActionCreators.removeCase,
+      onSnoozeUpdate: casesActionCreators.updateSnooze
+    },
+    dispatch
+  );
+
+type Props = ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+export const UnconnectedI90Table: React.FC<Props> = ({
+  caselist,
+  caseType,
+  isLoading,
+  toggleDetails,
+  updateSummaryData,
+  setError,
+  setNotification,
+  removeCase,
+  onSnoozeUpdate
+}) => {
+  if (caselist.length === 0 && isLoading) {
+    return null;
+  }
+
+  if (caselist.length === 0 && !isLoading) {
+    return <p>No cases found.</p>;
+  }
+
+  const numberOfColumns = 10;
```

Comment:
this seems very circumstantial that there are 10 columns for both the active and snoozed cases. I think it might be better to have this as a calculated field. 

we could define the headers as a list:
```
  const headers = [
    <th className="min"></th>,
    <th>Receipt Number</th>,
    <th>Case Age</th>,
    caseType === "active" && <th>Case Creation</th>,
    <th>Application Reason</th>,
    caseType === "active" && <th>Case Status</th>,
    caseType === "active" && <th>Case Substatus</th>,
    <th>Platform</th>,
    caseType === "snoozed" && <th>Problem</th>,
    caseType === "snoozed" && <th>Snoozed</th>,
    <th>Assigned</th>,
    caseType === "snoozed" && <th>SN Ticket #</th>,
    <th>Actions</th>,
  ].filter((header) => header !== false);
```
then update the render too
```
{headers.map((header, index) => <React.Fragment key={index}>{header}</React.Fragment>)}
```
and the call to caseDetails as
```
 <CaseDetails numberOfColumns={headers.length} rowData={r} />
```

Reply:
Yes, magic numbers are bad. Something feels weird about having an array of headers but not having an array of the actual content. Also, I think how you've defined the header list might not work perfectly since `caseType === "active"` or `caseType === "snoozed"` will still be an array element even if they evaluate to false. Will think on this.

Reply:
I think the call to filter should remove all the `false` values from the array. 

Maybe a better approach would be to add a test for active and snoozed to ensure that the row columns, header columns and value passed to CaseDetails are all equal

Reply:
Totally missed the `filter`, sorry.

I'm going to try chunking the cells out into their own components now and creating an array to iterate over for the table, which I think should accomplish the cell-by-cell validation via storybook (or snapshots) and also make the I90Table component more readable. Something like the following, although I'm still trying to think of best way to do the display logic.

```
const elements: Array<{ header: string; cell: React.FC<ElProps> }> = [
    {
      header: "",
      cell: ({ caseData }) => (
        <DetailToggle caseData={caseData} toggleDetails={toggleDetails} />
      )
    },
    {
      header: "Receipt Number",
      cell: ({ caseData }) => <ReceiptNumberLink caseData={caseData} />
    },
    {
      header: "Case Age",
      cell: ({ caseData }) => <CaseAge caseData={caseData} />,
      displayIf: caseType === "active"
    }
  ].filter(({ displayIf }) => displayIf === undefined || displayIf);
```